src/psyclone/transformations.py

```diff
@@ -1153,6 +1359,133 @@ def apply(self, nodes):
+    >>>     newschedule,memento=ltrans.apply(child)
+    >>>     schedule = newschedule
+    >>>
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # PARALLEL region
+    >>> newschedule, _ = rtrans.apply(schedule.children)
+    >>> newschedule.view()
+
+    '''
+    def __init__(self):
+        super(OMPParallelTrans, self).__init__()
+        from psyclone.psyGen import OMPParallelDirective
+        # Set the type of directive that the base class will use
+        self._pdirective = OMPParallelDirective
+
+    def __str__(self):
+        return "Insert an OpenMP Parallel region"
+
+    @property
+    def name(self):
+        ''' Returns the name of this transformation as a string.'''
+        return "OMPParallelTrans"
+
+    def _validate(self, node_list):
+        '''
+        Perform OpenMP-specific validation checks.
+
+        :param node_list: List of Nodes to put within parallel region.
+        :type node_list: list of :py:class:`psyclone.psyGen.Node`.
+        :raises TransformationError: if the target Nodes are already within \
+                                     some OMP parallel region.
+        '''
+        from psyclone.psyGen import OMPDirective
+
+        if node_list[0].ancestor(OMPDirective):
+            raise TransformationError("Error in OMPParallel transformation:" +
+                                      " cannot create an OpenMP PARALLEL " +
+                                      "region within another OpenMP region.")
+
+        # Now call the general validation checks
+        super(OMPParallelTrans, self)._validate(node_list)
+
+
+class ACCParallelTrans(ParallelRegionTrans):
+    '''
+    Create an OpenACC parallel region by inserting directives. This parallel
+    region *must* come after an enter-data directive (see `ACCDataTrans`). For
+    example:
+
+    >>> from psyclone.parse import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api="gocean1.0"
+    >>> filename="nemolite2d_alg.f90"
+    >>> ast,invokeInfo=parse(filename,api=api,invoke_name="invoke")
+    >>> psy=PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.psyGen import TransInfo
+    >>> t=TransInfo()
+    >>> ptrans = t.get_trans_name('ACCParallelTrans')
+    >>> dtrans = t.get_trans_name('ACCDataTrans')
+    >>>
+    >>> schedule=psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    >>> # Enclose everything within a single OpenACC PARALLEL region
+    >>> newschedule, _ = ptrans.apply(schedule.children)
+    >>> # Add an enter-data directive
+    >>> newschedule, _ = dtrans.apply(newschedule)
+    >>> newschedule.view()
+    '''
+    def __init__(self):
+        super(ACCParallelTrans, self).__init__()
+        from psyclone.psyGen import ACCParallelDirective
+        # Set the type of directive that the base class will use
+        self._pdirective = ACCParallelDirective
+
+    def __str__(self):
+        return "Insert an OpenACC Parallel region"
```

Comment:
description string and interface info?

Reply:
I've not documented the `__str__` methods because they are "special" and thus already defined.